src/logger.cr

```diff
@@ -0,0 +1,72 @@
+class Logger
```

Comment:
We could make `Logger` be `Logger(T)`, where `T` would be automatically inferred from the IO. In that way if you instantiate many loggers across your program, each with a different IO type, the `@io` instance variable won't be a big union that would involve a dispatch on each call. This is just something small, but I think it can improve performance a bit. A similar thing is done in [BufferedIO](https://github.com/manastech/crystal/blob/master/src/io/buffered_io.cr#L1). But no need to do this right now, and it can be changed later without affecting clients :-)